src/util/string_utils.cpp

```diff
@@ -156,13 +156,16 @@ std::string escape_non_alnum(const std::string &to_escape)
   std::ostringstream escaped;
   for(auto &ch : to_escape)
   {
+    const auto uch = static_cast<unsigned char>(ch);
     if(ch == '_')
       escaped << "__";
-    else if(isalnum(ch))
+    else if(isalnum(uch))
       escaped << ch;
     else
+    {
       escaped << '_' << std::hex << std::setfill('0') << std::setw(2)
-              << (unsigned int)ch;
+              << (unsigned int)uch;
```

Comment:
Since you’re changing this line anyway could you turn this into `static_cast` please?

Reply:
Now done as a `narrow` style cast instead as that is the preferred way to cast numerics according to the C++ core guidelines.

Reply:
@thomasspriggs I don't think this is a good in this instance as this is a widening cast, not a narrowing cast, so it's a bit confusing.

Reply:
Ok. I have swapped the conversion to `unsigned char` back to `static_cast`, because reinterpretation is the desired behaviour here. I have swapped the conversion from `unsigned char` to `int`, such that it is done using braced initialisation because this form of initialisation supports widening, but not narrowing. Therefore it makes it explicit that we are not performing narrowing.